frame/society/src/lib.rs

```diff
@@ -343,30 +348,134 @@ pub struct Bid<AccountId, Balance> {
+	/// The reward that the bidder has requested for successfully joining the society.
+	bid: Balance,
+	/// The tally of votes so far.
+	tally: Tally,
+	/// True if the skeptic was already punished for note voting.
+	skeptic_struck: bool,
+}
+
 /// A vote by a member on a candidate application.
 #[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
 pub enum BidKind<AccountId, Balance> {
-	/// The CandidateDeposit was paid for this bid.
+	/// The given deposit was paid for this bid.
 	Deposit(Balance),
 	/// A member vouched for this bid. The account should be reinstated into `Members` once the
 	/// bid is successful (or if it is rescinded prior to launch).
 	Vouch(AccountId, Balance),
 }
 
 impl<AccountId: PartialEq, Balance> BidKind<AccountId, Balance> {
-	fn check_voucher(&self, v: &AccountId) -> DispatchResult {
-		if let BidKind::Vouch(ref a, _) = self {
-			if a == v {
-				Ok(())
-			} else {
-				Err("incorrect identity".into())
-			}
-		} else {
-			Err("not vouched".into())
-		}
+	fn is_vouch(&self, v: &AccountId) -> bool {
+		matches!(self, BidKind::Vouch(ref a, _) if a == v)
 	}
 }
 
+pub type PayoutsFor<T, I> = BoundedVec<
+	(<T as frame_system::Config>::BlockNumber, BalanceOf<T, I>),
+	<T as Config<I>>::MaxPayouts,
+>;
+
+/// Information concerning a member.
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct MemberRecord {
+	rank: Rank,
+	strikes: StrikeCount,
+	vouching: Option<VouchingStatus>,
+	index: u32,
+}
+
+/// Information concerning a member.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, Default)]
+pub struct PayoutRecord<Balance, PayoutsVec> {
+	paid: Balance,
+	payouts: PayoutsVec,
+}
+
+pub type PayoutRecordFor<T, I> = PayoutRecord<
+	BalanceOf<T, I>,
+	BoundedVec<
+		(<T as frame_system::Config>::BlockNumber, BalanceOf<T, I>),
+		<T as Config<I>>::MaxPayouts,
+	>,
+>;
+
+/// Record for an individual new member who was elevated from a candidate recently.
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct IntakeRecord<AccountId, Balance> {
+	who: AccountId,
+	bid: Balance,
+	round: RoundIndex,
+}
+
+pub type IntakeRecordFor<T, I> =
+	IntakeRecord<<T as frame_system::Config>::AccountId, BalanceOf<T, I>>;
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct GroupParams<Balance> {
+	max_members: u32,
+	max_intake: u32,
+	max_strikes: u32,
+	candidate_deposit: Balance,
+}
+
+pub type GroupParamsFor<T, I> = GroupParams<BalanceOf<T, I>>;
+
+/// A vote by a member on a candidate application.
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum OldVote {
+	/// The member has been chosen to be skeptic and has not yet taken any action.
+	Skeptic,
+	/// The member has rejected the candidate's application.
+	Reject,
+	/// The member approves of the candidate's application.
+	Approve,
+}
+
 #[frame_support::pallet]
 pub mod pallet {
```

Comment:
I suggest setting a version on the pallet to make it clear this pallet code is consistent with the `MigrateToV2` migration